card/src/main/java/com/adyen/checkout/card/NewCardDelegate.kt

```diff
@@ -52,4 +82,110 @@ class NewCardDelegate(
     override fun isHolderNameRequired(): Boolean {
         return cardConfiguration.isHolderNameRequired
     }
+
+    override fun detectCardType(
+        cardNumber: String,
+        publicKey: String,
+        coroutineScope: CoroutineScope
+    ): List<DetectedCardType> {
+        Logger.d(TAG, "detectCardType")
+        if (cardNumber.length >= BinLookupConnection.REQUIRED_BIN_SIZE) {
+            val bin = cardNumber.substring(0, BinLookupConnection.REQUIRED_BIN_SIZE)
+            val hashedBin = Sha256.hashString(bin)
+
+            // if length is exactly the size, we call bin lookup API
+            if (cardNumber.length == BinLookupConnection.REQUIRED_BIN_SIZE) {
+                Logger.d(TAG, "Launching Bin Lookup")
+                coroutineScope.launch {
+                    val binLookupResponse = makeBinLookup(cardNumber, publicKey)
+                    if (binLookupResponse != null) {
+                        handleBinLookupResponse(hashedBin, binLookupResponse)
+                    }
+                }
+            }
+
+            // Check if we have a cached value
+            if (cachedBinLookup.containsKey(hashedBin)) {
+                Logger.d(TAG, "Returning cashed result.")
+                val cashedResult = cachedBinLookup[hashedBin]
+                if (cashedResult != null) {
+                    return cashedResult
+                }
+            }
+        }
+
+        // regexes logic from updateSupportedFilterCards()
+        return detectCardLocally(cardNumber)
+    }
+
+    private fun detectCardLocally(cardNumber: String): List<DetectedCardType> {
+        Logger.d(TAG, "detectCardLocally")
+        if (cardNumber.isEmpty()) {
+            return emptyList()
+        }
+        val supportedCardTypes = cardConfiguration.supportedCardTypes
+        val estimateCardTypes = CardType.estimate(cardNumber)
+        val detectedCardTypes = supportedCardTypes.filter { estimateCardTypes.contains(it) }
+        return detectedCardTypes.map { localDetectedCard(it) }
+    }
+
+    override fun localDetectedCard(cardType: CardType): DetectedCardType {
+        return DetectedCardType(
+            cardType,
+            isReliable = false,
+            showExpiryDate = true,
+            enableLuhnCheck = true,
+            cvcPolicy = getCvcPolicy(cardType.txVariant)
+        )
+    }
+
+    override fun getCvcPolicy(brand: String): Brand.CvcPolicy {
+        return when {
+            cardConfiguration.isHideCvc || noCvcBrands.contains(brand) -> Brand.CvcPolicy.HIDDEN
```

Comment:
`noCvcBrands` is a `Set<CardType>` while `brand` is `String`. Did you mean to do something like `noCvcBrands.any { it.txVariant == brand }`?

Reply:
Yeah I noticed this on the refactor, thanks for catching it :)